plugins/inputs/supervisor/README.md

```diff
@@ -0,0 +1,91 @@
+# Supervisor Input Plugin
+
+This plugin gather information about processes that running under supervisor using XML-RPC API
+
+Plugin minimum tested version: 3.3.2
+
+## Supervisor configuration
+
+This plugin needs TCP HTTP server to be enabled for collecting information. Here is example of
+`inet_http_server` section in supervisor config that will work with default plugin configuration.
+
+```ini
+[inet_http_server]
+port=127.0.0.1:9001
+```
+
+It is also recommended to setup basic authentication to http server as described [here](http://supervisord.org/configuration.html#inet-http-server-section-values).
+
+## Plugin configuration
+
+```toml
+[inputs.supervisor]
+  ## Url of supervisor's XML-RPC endpoint if basic auth enabled in supervisor http server,
+  ## than you have to add credentials to url (ex. http://login:pass@localhost:9001/RPC2)
+  # url="http://localhost:9001/RPC2"
+  ## Use supervisor identification string as server tag
+  # use_identification_tag = false
```

Comment:
I'm not sure if it makes sense to add "none" here. Even if it doesn't, I think a string setting is more descriptive than a bool.

This requires a corresponding change in parseProcessData and parseInstanceData. Where there is `if s.UseIdentTag {` we'll need `switch s.ServerTag {`

Reply:
Hi @niasar, thanks for updating this to be a string setting like I recommended.

Sorry to flip flop on what to do with this setting months later, but I was recently reminded that we have a convention for tags that identify the source of metric data. See https://github.com/influxdata/telegraf/blob/master/docs/developers/REVIEWS.md#source-tag

Instead of having a single tag "server" and a setting that lets the user choose if that value is the instance name or the host and port, it probably would be more useful not to have any setting for this and to always add a tag "instance" with the instance name and always add a "source" tag that has the hostname along with a "port" tag with the port.

This way the "server" tag won't be ambiguous. Users of the metrics down the road won't need to know how the plugin was configured to know whether "server" holds instance name or host/port. The source tag will also make the metrics more uniform compared to metrics from other plugins.

If someone doesn't want these tags, they can use the existing tagexclude setting to remove them. https://github.com/influxdata/telegraf/blob/master/docs/CONFIGURATION.md#modifiers

What do you think about this proposal? Maybe it makes sense to follow the convention and remove the ambiguity now so we don't have to live with it or fix it later?

Reply:
Not a problem, I've dropped server_tag from plugin configuration and added 3 separate tags(supervisor_host, supervisor_port, supervisor_id), that always will be included in plugins output.